benchmarks/benchmark/tools/locust-load-inference/locust-docker/locust-tasks/tasks.py

```diff
@@ -339,13 +341,16 @@ def grpc_infer(self):
         )
         logging.info(f"Prompt: {prompt}")
         #return values format is from the interceptor, which makes the actual call
-        output, ttft, response_time = self.stub.Decode(request)
-        logging.info(f"Response: {output}")
-
-        number_of_input_tokens = len(tokenizer.encode(prompt))
-        number_of_output_tokens = len(tokenizer.encode(output))
-        send_metrics(number_of_input_tokens, number_of_output_tokens, response_time,1, ttft)
+        try:
+            output, ttft, response_time = self.stub.Decode(request)
+            logging.info(f"Response: {output}")
 
+            number_of_input_tokens = len(tokenizer.encode(prompt))
+            number_of_output_tokens = len(tokenizer.encode(output))
+            send_metrics(number_of_input_tokens, number_of_output_tokens, response_time, 1, ttft)
+        except:
+            # Capture that a test was ran, but the request threw an exception
+            send_metrics(-1,-1,-1,0,-1)
```

Comment:
Wondering how doing -1 here will impact overall average data calculated in the MetricCollecter as it is averaged out over time. Do we discard the requests that throw an exception in those calculations?

Reply:
The metrics locust captures likely wouldnt be hit. As we would need to call `self.env.events.request.fire` for the request to be captures, and it's low probability that an exception would be thrown by the time it gets there. 

As far as scraping and averaging the metrics. That is true that it could miscalculate the averages. However, much of the same data is captured in the CSV, so these sentinel values could be filtered off. 

We may need to take another pass at the runner report. But I considered that out of scope.

Reply:
I was thinking more like we need to filter out the -1's in this calculation here: https://github.com/GoogleCloudPlatform/ai-on-gke/blob/01630c62fac8e5230c7cc0f8ff338e8fb3da8a5a/benchmarks/benchmark/tools/locust-load-inference/locust-docker/locust-tasks/custom_metric_aggregator.py#L36

Reason being, these avg metrics are also continuously scraped to cloud monitoring for real time metrics, and has also been useful in the autoscaling benchmarking work outside of the csv work. If we include the -1 values in this calculation, that data will be tainted. So I think filtering out those metrics in this calculation is important to implement within this initial PR to ensure we don't introduce data errors into other streams of data collection.

Reply:
Ah, I get what you're saying now. This should be circumnavigated since I pass in a `0` for request successful: https://github.com/GoogleCloudPlatform/ai-on-gke/blob/main/benchmarks/benchmark/tools/locust-load-inference/locust-docker/locust-tasks/custom_metric_aggregator.py#L15

I capture all the request metrics in a dictionary to make sure all the metrics of one request stay linked.

Reply:
Sweet, that lgtm!